tt_metal/impl/buffers/buffer.hpp

```diff
@@ -184,6 +185,8 @@ class Buffer {
 
     uint32_t num_pages() const { return this->size() / this->page_size(); }
 
+    void set_page_size(uint64_t page_size) { page_size_ = page_size; }
```

Comment:
should add checks here that the new page size doesnt cause the buffer to exceed previously allocated size

Reply:
yes, changed